src/components/FileViewer.jsx

```diff
@@ -12,17 +12,32 @@ class FileViewer extends React.Component {
   }
 
   componentDidMount () {
-    const intent = this.props.intent
-    cozy.client.intents.createService(intent, window)
-      .then(service => {
+    const { intentId } = this.props
+
+    let service, intent
+    cozy.client.intents.createService(intentId, window)
+      .then(_service => {
+        service = _service
```

Comment:
I don't like this variable naming.
Maybe `service` should be named `serviceAnchor` as its purpose is to be used to `terminate` it.
and `_service` should be named `service` as the function `createService` is resolved to a `service`.